### PR TITLE
misc(query): use local dispatcher for binary join when all metrics belong to same cluster

### PR DESCRIPTION
**Pull request by TanviBhavsar**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [x] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?

**Current behavior :** 
SinglePartitionPlanner uses `InProcessPlanDispatcher` for Binary Join

**New behavior :**
 Use local dispatcher when all metrics belong to same cluster

### CODE REVIEW

File: coordinator/src/main/scala/filodb.coordinator/queryplanner/SinglePartitionPlanner.scala [hunk 1]
```diff
@@ -39,38 +39,58 @@ class SinglePartitionPlanner(planners: Map[String, QueryPlanner],
     if(planner.isEmpty)  planners.values.head else planner.head
   }
 
-  private def materializeSimpleQuery(logicalPlan: LogicalPlan, qContext: QueryContext): ExecPlan = {
-    getPlanner(logicalPlan).materialize(logicalPlan, qContext)
-  }
+  private def getPlanner(binaryJoin: BinaryJoin) : Seq[QueryPlanner] = {
```
Comment (vishramachandran) [inline coordinator/src/main/scala/filodb.coordinator/queryplanner/SinglePartitionPlanner.scala:42]: Can you explain this function in plain english ? If lhs is not a binary join, then `getPlanner(lhs)` ... not understanding how this is compiling.

Comment (TanviBhavsar) [reply coordinator/src/main/scala/filodb.coordinator/queryplanner/SinglePartitionPlanner.scala]: I changed the overloaded `getPlanner` to `getBinaryJoinPlanners`. For BinaryJoin we need to call getPlanner on LHS and RHS and return the combined result.

**Review by vishramachandran — APPROVED** [review]
